src/libraries/System.Private.Xml/src/System/Xml/Serialization/ReflectionXmlSerializationReader.cs

```diff
@@ -399,17 +416,19 @@ private object GenerateEncodedMembersElement(XmlMembersMapping xmlMembersMapping
                 }
                 else
                 {
-                    unrecognizedElementSource = (_) =>
-                      {
-                          if (Reader.GetAttribute("id", null) != null)
-                          {
-                              ReadReferencedElement();
-                          }
-                          else
-                          {
-                              UnknownNode(p);
-                          };
-                      };
+                    unrecognizedElementSource = Wrapper;
```

Comment:
@agocke - is there a performance penalty for this pattern? This issue is still open - [https://github.com/dotnet/roslyn/issues/5835](https://github.com/dotnet/roslyn/issues/5835).

Or is the idea that since the lambda method already capturing `this`, there is no perf difference?

Reply:
Technically local functions are not method groups. This should produce the exact same IL as a lambda with the same capture semantics.